Shopify/cli
# Link uses api client config as fallback

### WHY are these changes introduced?
Last days users are reporting that the cli is returning this error when running  whatever `app`  command
```
╭─ error ─────────────────────────────────────────────────╮
│                                                         │
│  Fix a schema error in shopify.app.my-config.toml:      │
│  [                                                      │
│    {                                                    │
│      "code": "invalid_type",                            │
│      "expected": "string",                              │
│      "received": "undefined",                           │
│      "path": [                                          │
│        "name"                                           │
│      ],                                                 │
│      "message": "Required"                              │
│    },                                                   │
│    {                                                    │
│      "code": "invalid_type",                            │
│      "expected": "object",                              │
│      "received": "undefined",                           │
│      "path": [                                          │
│        "auth"                                           │
│      ],                                                 │
│      "message": "Required"                              │
│    },                                                   │
│    {                                                    │
│      "code": "invalid_type",                            │
│      "expected": "object",                              │
│      "received": "undefined",                           │
│      "path": [                                          │
│        "webhooks"                                       │
│      ],                                                 │
│      "message": "Required"                              │
│    },                                                   │
│    {                                                    │
│      "code": "invalid_type",                            │
│      "expected": "string",                              │
│      "received": "undefined",                           │
│      "path": [                                          │
│        "application_url"                                │
│      ],                                                 │
│      "message": "Required"                              │
│    },                                                   │
│    {                                                    │
│      "code": "invalid_type",                            │
│      "expected": "boolean",                             │
│      "received": "undefined",                           │
│      "path": [                                          │
│        "embedded"                                       │
│      ],                                                 │
│      "message": "Required"                              │
│    }                                                    │
│  ]                                                      │
│                                                         │
╰─────────────────────────────────────────────────────────╯
```

An example [here](https://github.com/Shopify/cli/issues/3618)

The issue was introduced in the version `3.56.0` by this [PR](https://github.com/Shopify/cli/pull/3411)

Most of the `app` commands runs the `link` logic under the hood to sync the local config with the remote one as a first step. The problem is that there are lot of apps that doesn't have any active app version so the final configuration written in the `toml` in these scenarios is invalid.

Before the versioned app config project was rolled out, a backfill process was run for the most important/installed apps. This process created a new app version using the API Client configuration to populate the `configuration` app modules. 
However, there are bunch of apps that the process didn't include in that first stage. In the following weeks the process will be executed for the rest of apps.

In the meantime we need a fix for any app that doesn't have any app version or the existing ones don't include any `configuration` app module



### WHAT is this pull request doing?
- `link` logic is using the API Client configuration content as a fallback in case the app doesn't have any app version


### How to test your changes?
- Run any `app` command for an old remote app that doesn't include any `configuration` app modules. They execution should succeed
- Run any `app` command for an old remote app that includes some `configuration` app modules. They execution should succeed




### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
